messages/src/lib.rs

```diff
@@ -1389,18 +1390,76 @@ impl<Q: QueryType> Decode for PartialBatchSelector<Q> {
     }
 }
 
+/// DAP protocol message representing an identifier for a collection.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct CollectionJobId([u8; Self::LEN]);
+
+impl CollectionJobId {
+    /// LEN is the length of a collection ID in bytes.
+    pub const LEN: usize = 16;
+}
+
+impl AsRef<[u8; Self::LEN]> for CollectionJobId {
+    fn as_ref(&self) -> &[u8; Self::LEN] {
+        &self.0
+    }
+}
+
+impl TryFrom<&[u8]> for CollectionJobId {
+    type Error = &'static str;
+
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        Ok(Self(value.try_into().map_err(|_| {
+            "byte slice has incorrect length for CollectionId"
+        })?))
+    }
+}
+
+impl FromStr for CollectionJobId {
+    type Err = Box<dyn Debug>;
```

Comment:
WDYT about using `anyhow::Error` as our "don't care about the type" error? It's about as expressive as a `Box<dyn Debug>`, we don't need to fiddle with boxing (ourselves), and it allows wrapping (so e.g. we could wrap each error condition with a helpful human-readable error message, if we wanted to).

Reply:
That would indeed work, but it brings its own awkwardness (`anyhow::Context::context` refuses to wrap `&str` so we'd have to mess with more error types) and it's not clear to me that it's sufficiently better than `Box<dyn Debug>` to justify the extra gymnastics.